docs/docs/migration/migration-from-v6-to-v7.md

```diff
@@ -136,38 +215,735 @@ It automatically installs the new versions of all `@comet` libraries, runs an ES
+
+### Add a custom `InternalLinkBlock`
+
+The `InternalLinkBlock` provided by `@comet/cms-site` is deprecated.
+Instead, implement your own `InternalLinkBlock`.
+This is needed for more flexibility, e.g., support for internationalized routing.
+
+### Add `aspectRatio` to `PixelImageBlock` and `Image`
+
+Previously, there was a default aspect ratio of `16x9`.
+This has repeatedly led to incorrectly displayed images.
+
+Now `aspectRatio` is required and must be added to `PixelImageBlock` and `Image`.
+**Consider which aspect ratio should be used.**
+
+Example:
+
+```diff
+ <PixelImageBlock
+   data={teaser}
+   layout="fill"
++  aspectRatio="16x9"
+ />
+```
+
+### Make relative DAM URLs work
+
+This requires the following change (depending on which router you use):
+
+#### Pages Router
+
+```diff
+// next.config.js
+
+const nextConfig = {
+    rewrites: async () => {
+        if (process.env.NEXT_PUBLIC_SITE_IS_PREVIEW === "true") return [];
+        var rewrites = await require("./preBuild/build/preBuild/src/createRewrites").createRewrites();
+-       return rewrites;
++       return [
++           ...rewrites,
++           {
++               source: "/dam/:path*",
++               destination: process.env.API_URL + "/dam/:path*",
++           },
++       ];
+    },
+    // ...
+```
+
+#### App Router
+
+```diff
+// middleware.ts
+
+export async function middleware(request: NextRequest) {
++   if (request.nextUrl.pathname.startsWith("/dam/")) {
++       return NextResponse.rewrite(new URL(`${process.env.API_URL_INTERNAL}${request.nextUrl.pathname}`));
++   }
+    // ...
+```
+
+### Switch to Next.js preview mode
+
+#### Requires following changes to site:
+
+-   Import `useRouter` from `next/router` (not exported from `@comet/cms-site` anymore)
+-   Import `Link` from `next/link` (not exported from `@comet/cms-site` anymore)
+-   Remove preview pages (pages in `src/pages/preview/` directory which call `createGetUniversalProps` with preview parameters)
+-   Remove `createGetUniversalProps`
+    -   Just implement `getStaticProps`/`getServerSideProps` (Preview Mode will SSR automatically)
+    -   Get `previewData` from `context` and use it to configure the GraphQL Client
+-   Add `SitePreviewProvider` to `App` (typically in `src/pages/_app.tsx`)
+-   Provide a protected environment for the site
+    -   Make sure that a Authorization-Header is present in this environment
+    -   Add a Next.JS API-Route for the site preview (eg. `/api/site-preview`)
+    -   Call `getValidatedSitePreviewParams()` in the API-Route (calls the API which checks the Authorization-Header with the submitted scope)
+    -   Use the `path`-part of the return value to redirect to the preview
+
+#### Requires following changes to admin:
+
+-   The `SitesConfig` must provide a `sitePreviewApiUrl`
+
+### TODO: GraphQL fetch client
+
+// TODO ?
+
+plenty-cougars-warn.md
+
+### TODO: New technique for blocks to load additional data at page level
+
+// TODO ?
```

Comment:
@nsams Should we document this in the migration guide? What do you think

Reply:
that's nothing you have to use when upgrading. You should only know that this feature exists - but that is not the what a migration guide is good for(?)

Reply:
> but that is not the what a migration guide is good for(?)

Technically not. But it's the only document many devs ever actually look at when updating to a new major version.

ATM we don't have any docs on how to use this feature. The changeset is also very brief. So nobody will know this feature exists